basis/windows/dragdrop-listener/dragdrop-listener.factor

```diff
@@ -84,6 +73,10 @@ SYMBOL: +listener-dragdrop-wrapper+
 >>
 
 : dragdrop-listener-window ( -- )
-    world get handle>> hWnd>> dup <listener-dragdrop>
-    +listener-dragdrop-wrapper+ get-global com-wrap
-    [ RegisterDragDrop check-ole32-error ] with-com-interface ;
+    world get dup <listener-dragdrop>
+    +listener-dragdrop-wrapper+ get-global com-wrap [
+        [ handle>> hWnd>> ] dip
+        2dup RegisterDragDrop dup E_OUTOFMEMORY =
```

Comment:
Why do you check ``E_OUTOFMEMORY`` here  and then call ``ole-initialize`` and call it again?

Reply:
Because that's exactly the error code it returns the first time if `ole-initialize` was never called.

`ole-initialize` needs to be called once per application run, before OLE-related functions start working. I didn't want to call it in a startup hook in case we never actually use `RegisterDragDrop`. And I couldn't assume nobody else called it before, so I need to call it at some point.

The way I did it here, it will be called only once, and the rest of the time the first `RegisterDragDrop` will succeed.

Theoretically, it is harmless to call it every time before each `RegisterDragDrop`, but it seemed a bit sloppy, since a one-time operation becomes all-the-time operation.

How do you usually do these things?